UnitySDK/Assets/ML-Agents/Scripts/Grpc/GrpcExtensions.cs

```diff
@@ -38,15 +38,12 @@ public static AgentInfoProto ToProto(this AgentInfo ai)
                 agentInfoProto.ActionMask.AddRange(ai.actionMasks);
             }
 
-            foreach (var obs in ai.visualObservations)
+            foreach (var obs in ai.compressedObservations)
             {
-                using (TimerStack.Instance.Scoped("encodeVisualObs"))
-                {
-                    agentInfoProto.VisualObservations.Add(
-                        ByteString.CopyFrom(obs.EncodeToPNG())
-                    );
-                }
+                // TODO pass compression type and dimensions separately.
+                agentInfoProto.VisualObservations.Add(ByteString.CopyFrom(obs.Data));
```

Comment:
Could you add a Timer/Profiler around this.  Would be interesting to keep track of this, or maybe one level up, for conversion of all Sensors --> Proto

Reply:
Added timers in GrpcCommunicator around the whole AgentInfo.ToProto loop, and also around the bodies of the ISensor methods.